src/util/finite_field.h

```diff
@@ -0,0 +1,195 @@
+   *
+   * For fields of odd size, there is always a unique representative of
+   * smallest absolute value.
+   *
+   * For GF(2), the multiplicative identity is represented as "1", not "-1".
+   * */
+  Integer toSignedInteger() const;
+  /* Return (binary) string representation. */
+  std::string toString(unsigned int base = 2) const;
+
+  /* Return hash value. */
+  size_t hash() const;
+
+  /* -----------------------------------------------------------------------
+   ** Operators
+   * ----------------------------------------------------------------------- */
+
+  /* (Dis)Equality --------------------------------------------------------- */
+
+  /* Return true if this is equal to 'y'. */
+  bool operator==(const FiniteField& y) const;
+
+  /* Return true if this is not equal to 'y'. */
+  bool operator!=(const FiniteField& y) const;
+
+  /* Unsigned Inequality --------------------------------------------------- */
+
+  /* Return true if this is unsigned less than bit-vector 'y'. */
+  bool operator<(const FiniteField& y) const;
+
+  /* Return true if this is unsigned less than or equal to bit-vector 'y'. */
+  bool operator<=(const FiniteField& y) const;
+
+  /* Return true if this is unsigned greater than bit-vector 'y'. */
+  bool operator>(const FiniteField& y) const;
+
+  /* Return true if this is unsigned greater than or equal to bit-vector 'y'. */
+  bool operator>=(const FiniteField& y) const;
+
+  /* Arithmetic operations ------------------------------------------------- */
+
+  /* Return a bit-vector representing the addition (this + y). */
+  FiniteField operator+(const FiniteField& y) const;
+
+  /* Return a bit-vector representing the subtraction (this - y). */
+  FiniteField operator-(const FiniteField& y) const;
+
+  /* Return a bit-vector representing the negation of this. */
+  FiniteField operator-() const;
+
+  /* Return a bit-vector representing the multiplication (this * y). */
+  FiniteField operator*(const FiniteField& y) const;
+
+  /* Return a bit-vector representing the division (this / y). */
+  FiniteField operator/(const FiniteField& y) const;
+
+  /* Reciprocal. Crashes on 0. */
+  FiniteField recip() const;
+
+  /* -----------------------------------------------------------------------
+   ** Static helpers.
+   * ----------------------------------------------------------------------- */
+
+  /* Create zero bit-vector of given size. */
+  static FiniteField mkZero(const Integer& modulus);
+
+  /* Create bit-vector representing value 1 of given size. */
+  static FiniteField mkOne(const Integer& modulus);
+
+ private:
+  /**
+   * Class invariants:
+   *  - no overflows: d_value < d_modulus
+   *  - no negative numbers: d_value >= 0
+   */
+
+  Integer d_size;
+  Integer d_value;
+
+}; /* class FiniteField */
+
+struct FiniteFieldSize
+{
+  Integer d_size;
+  FiniteFieldSize(Integer size) : d_size(size) {}
+  operator Integer() const { return d_size; }
+  inline bool operator==(const FiniteFieldSize& y) const
```

Comment:
```suggestion
  inline bool operator==(const FiniteFieldSize& y) const
```

[We don't use the `inline` specifier unless there is a good reason to do so.](https://github.com/cvc5/cvc5/wiki/Code-Style-Guidelines#inline-functions)

Reply:
:+1: